crates/sqllogictest/src/schedule.rs

```diff
@@ -105,103 +125,116 @@ impl Schedule {
-
-            let engine = load_engine_runner(engine_type, cfg_tbl.clone()).await?;
-
-            if engines.insert(name.clone(), engine).is_some() {
-                return Err(anyhow!("Duplicate engine '{name}'"));
-            }
-        }
+#[cfg(test)]
+mod tests {
+    use crate::engine::EngineType;
+    use crate::schedule::ScheduleConfig;
 
-        Ok(engines)
-    }
+    #[test]
+    fn test_deserialize_schedule_config() {
+        let input = r#"
+            [engines]
+            df = { type = "datafusion" }
 
-    fn parse_steps(table: &TomlTable) -> anyhow::Result<Vec<Step>> {
-        let steps_val = table
-            .get("steps")
-            .with_context(|| "Schedule file must have a 'steps' array")?;
+            [[steps]]
+            engine = "df"
+            slt = "test.slt"
+        "#;
 
-        let steps: Vec<Step> = steps_val
-            .clone()
-            .try_into()
-            .with_context(|| "Failed to deserialize steps")?;
+        let config: ScheduleConfig = toml::from_str(input).unwrap();
 
-        Ok(steps)
+        assert_eq!(config.engines.len(), 1);
+        assert!(config.engines.contains_key("df"));
+        assert_eq!(config.engines["df"].engine_type, EngineType::Datafusion);
+        assert_eq!(config.steps.len(), 1);
+        assert_eq!(config.steps[0].engine, "df");
+        assert_eq!(config.steps[0].slt, "test.slt");
     }
-}
-
-#[cfg(test)]
-mod tests {
-    use toml::Table as TomlTable;
-
-    use crate::schedule::Schedule;
 
     #[test]
-    fn test_parse_steps() {
+    fn test_deserialize_multiple_steps() {
         let input = r#"
+            [engines]
+            datafusion = { type = "datafusion" }
+
             [[steps]]
             engine = "datafusion"
             slt = "test.slt"
 
             [[steps]]
-            engine = "spark"
+            engine = "datafusion"
             slt = "test2.slt"
         "#;
 
-        let tbl: TomlTable = toml::from_str(input).unwrap();
-        let steps = Schedule::parse_steps(&tbl).unwrap();
+        let config: ScheduleConfig = toml::from_str(input).unwrap();
 
-        assert_eq!(steps.len(), 2);
-        assert_eq!(steps[0].engine, "datafusion");
-        assert_eq!(steps[0].slt, "test.slt");
-        assert_eq!(steps[1].engine, "spark");
-        assert_eq!(steps[1].slt, "test2.slt");
+        assert_eq!(config.steps.len(), 2);
+        assert_eq!(config.steps[0].engine, "datafusion");
+        assert_eq!(config.steps[0].slt, "test.slt");
+        assert_eq!(config.steps[1].engine, "datafusion");
+        assert_eq!(config.steps[1].slt, "test2.slt");
     }
 
     #[test]
-    fn test_parse_steps_empty() {
+    fn test_deserialize_with_extra_fields() {
+        // Test forward-compatibility with extra fields (for PR #1943)
         let input = r#"
+            [engines]
+            df = { type = "datafusion", catalog_type = "rest", some_future_field = "value" }
```

Comment:
```suggestion
            df = { type = "datafusion", 
            catalog = {
               type = "rest",
               props = {
                  // A hashmap of properties.
                  url = "http://localhost:9000",
                  use_x = "true",
               }
              }
            }       
```
This is a complete example, and then we could use catalog loader to load a catalog.